doc/source/data/transforming-datasets.rst

```diff
@@ -78,34 +80,338 @@ express your customized business logic in transformations. Here we will focus on
 :meth:`.map_batches() <ray.data.Dataset.map_batches>` as it's the primary mapping
 API in Datasets.
 
-A UDF can be a function or a callable class, which has the following input/output:
+A batch UDF can be a function or, if using the actor compute strategy, a callable class.
```

Comment:
The function can also use actor compute strategy, right?
I think it's better to leave compute strategy as separate one (as covered below).

Reply:
>The function can also use actor compute strategy, right?

Yes, this is saying "a batch UDF can be a function (for either compute strategy) or (if using the actor compute strategy) a callable class." That's what the "A or, if X, B" sentence structure means.

>I think it's better to leave compute strategy as separate one (as covered below).

This is an intro section for "Writing UDFs", so I think that quickly mentioning what a UDF can be before diving into the batch formats is prudent, and I think that we should have a dedicated callable class subsection in this section.

I do think that we should add a reference link here to the callable class and compute strategy sections here though.